### PR TITLE
Remove empty else condition

### PR DESCRIPTION
**Pull request by pbardea**
First attempt at #14

### CODE REVIEW

File: test/if.spec.js [hunk 1]
```diff
@@ -59,7 +59,25 @@ describe('If', () => {
...
+            const a = new cgl.Variable(cgl.Qualifier.In, cgl.Type.Bool, 'a');
+            const b = new cgl.Variable(cgl.Qualifier.In, cgl.Type.Bool, 'b');
+            const ifStmt = new cgl.If(
+                new cgl.Reference(a),
+                new cgl.Block([
+                    new cgl.Statement(
+                        new cgl.Assignment(new cgl.Reference(a), new cgl.Reference(b))
+                    )
+                ]),
+                new cgl.Block()
```
Comment (pbardea) [inline test/if.spec.js]: Also added this test, for the case where the user explicitly writes an empty block.

Unsure if the user would expect an else block in this case... It would be weird to do this though. And the work-around would be setting the default to `null` rather than an empty block. I think the current implementation is best, but just wanted to bring it up.

Comment (davepagurek) [reply test/if.spec.js]: I think we can just call this a "compiler optimization."

File: test/block.spec.js [hunk 2]
```diff
@@ -27,7 +27,7 @@ describe('Block', () => {
         });
     });
 
-    describe('dependencies', () => {
+    describe('source', () => {
```
Comment (pbardea) [inline test/block.spec.js]: @davepagurek This looks like it should have been source rather than dependencies. Can you confirm just to make sure?

File: test/block.spec.js [hunk 3]
```diff
@@ -51,4 +51,29 @@ describe('Block', () => {
             expect(block.source()).to.equalIgnoreSpaces('{}');
         });
     });
+
+    describe('isEmpty', () => {
```
Comment (davepagurek) [inline test/block.spec.js]: 👌 nice!

File: test/block.spec.js (hunk 2 above)
Comment (davepagurek) [reply test/block.spec.js]: Ah yep you're right
